pallets/order-book/src/lib.rs

```diff
@@ -582,6 +587,97 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Fill an existing order, based on the provided ratio.
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::Weights::fill_order_partial())]
+		pub fn fill_order_partial(
+			origin: OriginFor<T>,
+			order_id: T::OrderIdNonce,
+			buy_amount: T::Balance,
+		) -> DispatchResult {
```

Comment:
Change request: 

There's a lot of code duplication between `fill_order_full` and `fill_order_partial`. We should move this extrinsic's scope to a helper function which is called both from `fill_order_full` and `fill_order_partial`.

Reply:
Yup, already was looking into that.

Reply:
Done.